src/rules/preferConstRule.ts

```diff
@@ -0,0 +1,219 @@
+}
+
+class PreferConstWalker extends Lint.BlockScopeAwareRuleWalker<{}, ScopeInfo> {
+    public createScope() {
+        return {};
+    }
+
+    public createBlockScope() {
+        return new ScopeInfo();
+    }
+
+    public onBlockScopeEnd() {
+        const seenLetStatements: { [startPosition: string]: boolean } = {};
+        for (const usage of this.getCurrentBlockScope().getConstCandiates()) {
+            let fix: Lint.Fix;
+            if (!usage.reassignedSibling && !seenLetStatements[usage.letStatement.getStart().toString()]) {
+                // only fix if all variables in the `let` statement can use `const`
+                const replacement = new Lint.Replacement(usage.letStatement.getStart(), "let".length, "const");
+                fix = new Lint.Fix(Rule.metadata.ruleName, [replacement]);
+                seenLetStatements[usage.letStatement.getStart().toString()] = true;
+            }
+            this.addFailure(this.createFailure(
+                usage.identifier.getStart(),
+                usage.identifier.getWidth(),
+                Rule.FAILURE_STRING_FACTORY(usage.identifier.text),
+                fix,
+            ));
+        }
+    }
+
+    protected visitBinaryExpression(node: ts.BinaryExpression) {
+        if (isAssignmentOperator(node.operatorToken)) {
+            this.handleLHSExpression(node.left);
+        }
+        super.visitBinaryExpression(node);
+    }
+
+    protected visitEndOfFileToken(node: ts.Node) {
+        this.onBlockScopeEnd();
+        super.visitEndOfFileToken(node);
+    }
+
+    protected visitPrefixUnaryExpression(node: ts.PrefixUnaryExpression) {
+        this.handleUnaryExpression(node);
+        super.visitPrefixUnaryExpression(node);
+    }
+
+    protected visitPostfixUnaryExpression(node: ts.PostfixUnaryExpression) {
+        this.handleUnaryExpression(node);
+        super.visitPostfixUnaryExpression(node);
+    }
+
+    protected visitVariableDeclaration(node: ts.VariableDeclaration) {
+        this.getCurrentBlockScope().currentVariableDeclaration = node;
+        super.visitVariableDeclaration(node);
+        this.getCurrentBlockScope().currentVariableDeclaration = null;
+    }
+
+    protected visitIdentifier(node: ts.Identifier) {
+        if (this.getCurrentBlockScope().currentVariableDeclaration != null) {
+            const declarationList = this.getCurrentBlockScope().currentVariableDeclaration.parent;
+            if (isNodeFlagSet(declarationList, ts.NodeFlags.Let) && !isNodeFlagSet(declarationList.parent, ts.NodeFlags.Export)) {
+                if (this.isVariableDeclaration(node)) {
+                    this.getCurrentBlockScope().addVariable(node, declarationList);
+                }
+            }
+        }
+        super.visitIdentifier(node);
+    }
+
+    private handleLHSExpression(node: ts.Expression) {
+        node = unwrapParentheses(node);
+        if (node.kind === ts.SyntaxKind.Identifier) {
+            this.markAssignment(<ts.Identifier> node);
```

Comment:
it is, let's enable it :D https://palantir.github.io/tslint/rules/no-angle-bracket-type-assertion/